project/src/pages/room-page/room-page.tsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import Reviews from '../../components/reviews/reviews';
+import Map from '../../components/map/map';
+import { Offer, Offers } from '../../types/offers';
+import { getRatingStyle } from '../../utils';
+import OffersList from '../../components/offers-list/offers-list';
+
+function RoomPage(props: {offers: Offers}): JSX.Element{
+  const param = useParams();
+  const paramsId = Number(param.id);
+
+  const offersItem: Offer = paramsId ? props.offers.filter((currentOffer) => currentOffer.id === paramsId)[0] : props.offers[0];
+
+  const city = props.offers[0].city.location;
```

Comment:
нужно брать локацию у текущего оффера, который пользователь открыл на этой странице, а не первого из списка
```suggestion
  const city = props.currentOffer.city.location;
```

Reply:
переменная называется город, а в ней лежат координаты города — это плохо

переменная должна отвечать на вопрос что в ней лежит — `cityLocation`